### PR TITLE
bump: :tools tree-sitter

### PR DESCRIPTION
**Pull request by jeetelongname**
- Bump the module
- add support for
  - haskell
  - lua
  - erlang
  - ess-r
  - yaml
 - as well as docs for all

-----
- [x] I searched the issue tracker and this hasn't been PRed before.
- [ ] My changes are not on [the do-not-PR list](https://doomemacs.org/d/do-not-pr) for this project.
- [x] My commits conform to [the git conventions](https://doomemacs.org/d/git-conventions).
- [ ] My changes are visual; I've included before and after screenshots.
- [ ] I am blindly checking these off.
- [x] Any relevant issues or PRs have been linked to.
- [ ] This a draft PR; I need more time to finish it.

### CODE REVIEW

**Comment by iyefrat** [comment]
> add haskell support

Thanks yeet!

**Comment by jeetelongname** [comment]
The highlighting is interesting, but it has grown on me so I have left it as is

**Review by hlissner — CHANGES_REQUESTED** [review]
LGTM, but would you mind re-bumping the :tools tree-sitter module? All three packages have new commits.

**Comment by jeetelongname** [comment]
Sure, once i have emacs set up on my new machine ill bump the module again

File: modules/lang/ess/config.el [hunk 1]
```diff
@@ -23,6 +23,9 @@
   (when (modulep! +lsp)
     (add-hook 'ess-r-mode-local-vars-hook #'lsp! 'append))
 
+ (when (modulep! +tree-sitter)
+      (add-hook 'ess-r-mode-local-vars-hook #'tree-sitter! 'append))
```
Comment (hlissner) [inline modules/lang/ess/config.el:27]: Please correct the indentation.

File: modules/lang/lua/config.el [hunk 2]
```diff
@@ -71,7 +74,10 @@ lua-language-server.")
     tab-width 2
     ;; Don't treat autoloads or sexp openers as outline headers, we have
     ;; hideshow for that.
-    outline-regexp "[ \t]*;;;;* [^ \t\n]"))
+    outline-regexp "[ \t]*;;;;* [^ \t\n]")
+
+  (when (modulep! +tree-sitter)
+    (add-hook! 'fennel-mode-local-vars-hook 'tree-sitter!)))
```
Comment (hlissner) [inline modules/lang/lua/config.el:80]: Please append this hook, use `add-hook` instead, and sharp-quote `tree-sitter!`:

```elisp
(add-hook 'fennel-mode-local-vars-hook #'tree-sitter! 'append)
```

**Review by hlissner — CHANGES_REQUESTED** [review]

**Comment by jeetelongname** [comment]
@hlissner Done and Done

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
Great work, as always! Thanks again for your help!

**Pull request merged**